oauthenticator/globus.py

```diff
@@ -109,24 +119,42 @@ def _scope_default(self):
             'urn:globus:auth:scope:transfer.api.globus.org:all'
         ]
 
-    globus_local_endpoint = Unicode(help="""If Jupyterhub is also a Globus endpoint, its endpoint id can
-                be specified here.""").tag(config=True)
+    allow_refresh_tokens = Bool(
+        help="""Allow users to have Refresh Tokens. If Refresh Tokens are not
+        allowed, users must use regular Access Tokens which will expire after
+        a set time. Set to False for increased security, True for increased
+        convenience."""
+    ).tag(config=True)
+
+    def _allow_refresh_tokens_default(self):
+        return True
+
+    globus_local_endpoint = Unicode(help="""If Jupyterhub is also a Globus
+    endpoint, its endpoint id can be specified here.""").tag(config=True)
 
     def _globus_local_endpoint_default(self):
         return os.getenv('GLOBUS_LOCAL_ENDPOINT', '')
 
     def pre_spawn_start(self, user, spawner):
-        """Add tokens to the spawner whenever the spawner starts a notebook. Token names are translated to conform
-        to bash-style variable names. For example, 'transfer.api.globus.org' becomes 'TRANSFER_API_GLOBUS_ORG'"""
-        env_tokens = {
-            self.bashify_variable_name(name): token for name, token in user.auth_state['transfer_tokens'].items()
-        }
+        """Add tokens to the spawner whenever the spawner starts a notebook.
+        This will allow users to create a transfer client:
+        globus-sdk-python.readthedocs.io/en/stable/tutorial/#tutorial-step4
+        """
         if self.globus_local_endpoint:
-            spawner.environment.update({'GLOBUS_LOCAL_ENDPOINT': self.globus_local_endpoint})
-        spawner.environment.update(env_tokens)
-
-    def bashify_variable_name(self, var_name):
-        return var_name.replace('.', '_').replace('-', '_').upper()
+            spawner.environment.update(
+                {'GLOBUS_LOCAL_ENDPOINT': self.globus_local_endpoint}
+            )
+        if user.auth_state.get('globus_data'):
+            globus_data = base64.b64encode(
+                pickle.dumps(user.auth_state['globus_data'])
+            )
+            spawner.environment['GLOBUS_DATA'] = globus_data
```

Comment:
Everything the user needs should be in the `GLOBUS_DATA` environment variable when they start up their notebook. Now they'll need to decode and un-pickle the data before using them, instead of just pulling the access_tokens directly from the environment, but users shouldn't need to visit an external website for anything.